src/showcase/app/maps-leaflet/maps-leaflet-examples/leaflet-map-examples/leaflet-map-example/leaflet-map-example.component.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  Canvas,
+  circle,
+  latLng,
+  LatLngBounds,
+  LayerGroup,
+  layerGroup,
+  LeafletMouseEvent,
+  MapOptions,
+  polygon,
+  popup,
+  tileLayer,
+  Util,
+} from 'leaflet';
+
+export let mapOptions: MapOptions = {
+  zoom: 12,
+  center: latLng(46.948094, 7.451563),
+};
+
+export let layersControlConfig: LayersControl = {
+  baseLayers: [
+    {
+      title: 'OSM',
+      visible: true,
+      layer: tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+        maxZoom: 19,
+        attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+      }),
+    },
+  ],
+  overLays: [
+    {
+      title: 'AGOL',
+      visible: true,
+      layer: featureLayer({
+        url:
+          'https://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/Hazards_Uptown_Charlotte/FeatureServer/0',
+        renderer: new Canvas(),
+        useCors: true,
+      }).bindPopup((l: any) => {
+        return Util.template(
+          '<p>Standort <strong>{NAME}</strong> in {GEMEINDE}.',
+          l.feature.properties
+        );
+      }),
+    },
+    {
+      title: 'Shapes',
+      visible: true,
+      layer: layerGroup([
+        polygon(
+          [
+            [46.94469, 7.440515],
+            [46.945798, 7.441306],
+            [46.94634, 7.441016],
+            [46.94608, 7.439305],
+            [46.945443, 7.439611],
+          ],
+          {
+            color: 'red',
+          }
+        ).bindPopup(`<b>test</p>`),
+        circle([46.948212, 7.455189], {
+          color: 'red',
+          fillColor: '#f03',
+          fillOpacity: 0.2,
+          radius: 200,
+        }).bindPopup('sample'),
+      ]),
+    },
+  ],
+};
+
+@Component({
+  selector: 'sbb-leaflet-map-example',
+  templateUrl: './leaflet-map-example.component.html',
+  styleUrls: ['./leaflet-map-example.component.scss'],
+})
+export class LeafletMapExampleComponent implements AfterViewChecked {
+  private _map: L.Map;
+  private _myPointLayerGroup: LayerGroup<any>;
+
+  @ViewChild('leafletMap', { static: true })
+  private _leafLetMap: LeafletMapComponent;
+
+  public options = mapOptions;
+  public lc = layersControlConfig;
+
+  private _createLayerGroup(myPoints: [number, number][]): LayerGroup<any> {
+    const l = new LayerGroup();
+    myPoints.forEach((p) => {
+      l.addLayer(circle(p, { color: 'blue', radius: 10 }).bindPopup('Hi, I am a point.'));
+    });
+    return l;
+  }
+
+  private _removeLayerFromMap(l: any) {
+    if (l) {
+      this._leafLetMap.removeLayerFromMap(l);
+    }
+  }
+
+  private _generatePoints(): [number, number][] {
+    const pointArray = [];
+    for (let i = 1; i < 1000; i++) {
+      pointArray.push([this._randomNumber(45.8, 47.7), this._randomNumber(5.9, 10.5)]);
+    }
+    return pointArray;
+  }
+
+  ngAfterViewChecked() {
+    // get new mapsize when "examples"-tab is clicked
+    if (this._map) {
+      setTimeout(() => this._map.invalidateSize(), 300);
```

Comment:
Why 300 milliseconds? And why is this not called in mapReady?

Reply:
This workaround is necessary because the leaflet map gets its size when it is added to the DOM. At this point it is not yet displayed, but only when the user clicks the "Examples" tab. This leads to not enough tiles being loaded when the map is actually displayed. All required tiles will not be loaded until you call `invalidateSize()`. So this is not a problem of the Leaflet-map-component itself, but only in the showcase.

Is there any other way when this method can be called? For example can I get a notification when the user switches to the Examples tab?

Translated with www.DeepL.com/Translator (free version)

Reply:
I see. No, not easily.
Does it also work if you use `Promise.resolve().then(() => this._map.invalidateSize())`? We generally use Promise.resolve if something should be invoked in the next "tick".

Reply:
I tested it out - I think it doesn't work. When I use `Promise.resolve().then(() => this._map.invalidateSize())` the app does not respond anymore. (But the code in the callback method is still executed.)
When I do not run it with a timeout, the map loads the tile only if the user integrates with the app.

Reply:
Ok, so it won't work with a microtask. One more thing: Does it work with just `setTimeout(() => this._map.invalidateSize())`?
Either way it would probably be a good thing to document the use case in detail in the code, so consumers can understand why it is necessary.

Reply:
yep, this works! 😃 thanks for the tip!

Reply:
`Promise.resolve().then` schedules a microtask and `setTimeout` schedules a macrotask. Usually the difference doesn't matter too much, but sometimes it is relevent for waiting for other "work" on the thread.